src/source-control/GitSourceControl.ts

```diff
@@ -326,9 +339,10 @@ export class GitSourceControl implements SourceControl {
 
       queue.push(...commitToBeRebased.childCommits);
 
-      // We now would want to use this new commit hash as the target
-      // FIXME: This will turn a commit tree into a linear stack!
-      targetCommitHash = newCommitOid.tostrS();
+      // We now would want to use this new commit hash as the target for all of it's children
```

Comment:
Nit: rephrasing for clarity+grammar (imo, feel free to change further)

```suggestion
      // Register new commit as the rebase target for all children
```